core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go

```diff
@@ -36,7 +36,7 @@ func NewAddService(serviceNetwork service_network.ServiceNetwork, runtimeValueSt
 					IsOptional:        false,
 					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
 					Validator: func(value starlark.Value) *startosis_errors.InterpretationError {
-						return builtin_argument.NonEmptyString(value, ServiceNameArgName)
```

Comment:
Do you need to validate it twice here? If the validation was cheap, I'd say it doesn't hurt, but given that you have to write 20 lines of code here, I'd just keep the `NonEmptyString` default validator here.
The service names are going to be validated at the validation step anyway

Reply:
@gbouv I'm thinking that it's a better practice to move the validation as early stage as possible, i.e. interpretation stage as it provides the result as soon as possible. But it does have the implication that we then validate twice. If you feel it's a tradeoff worth making I can also go with removing it. Let me know.

Reply:
Generally speaking I'd agree that failing as early as possible is a good default strategy, but if it comes at the cost of adding 20 lines of code, given that it adds no value at all (interpretation is lightening fast), I would personally not do it.

Reply:
@gbouv Cool, I'm open to remove it. But for the record, I don't agree that it doesn't add any value, as it promotes the best practice of applying validation as early as possible, which can be useful particularly if the validation stage takes a long time to complete. However, there are always tradeoffs to be made and I'm not rigid about this one.